pkgs/misc/vim-plugins/update.py

```diff
@@ -24,12 +24,59 @@
 from urllib.parse import urljoin, urlparse
 from tempfile import NamedTemporaryFile
 
-ATOM_ENTRY = "{http://www.w3.org/2005/Atom}entry"
-ATOM_LINK = "{http://www.w3.org/2005/Atom}link"
-ATOM_UPDATED = "{http://www.w3.org/2005/Atom}updated"
+ATOM_ENTRY = "{http://www.w3.org/2005/Atom}entry" #" vim gets confused here
+ATOM_LINK = "{http://www.w3.org/2005/Atom}link" #"
+ATOM_UPDATED = "{http://www.w3.org/2005/Atom}updated" #"
 
 ROOT = Path(__file__).parent
 
+import time
+from functools import wraps
+
+
+def retry(ExceptionToCheck, tries=4, delay=3, backoff=2, logger=None):
```

Comment:
Can you add type hints here as well?

Reply:
```suggestion
def retry(ExceptionToCheck: Exception, tries: int = 4, delay : int = 3, backoff: int = 2):
```

Reply:
All the type hints are untested, so you may want to check with mypy.

Reply:
Unfortunately `Exception` didn't work, maybe python special cases that? mypy complained:

```
update.py:58: error: Exception type must be derived from BaseException
update.py:81: error: Argument 1 to "retry" has incompatible type "Type[URLError]"; expected "Exception"
update.py:94: error: Argument 1 to "retry" has incompatible type "Type[URLError]"; expected "Exception"
```

So I used "Any" instead (which also respects that you could theoretically pass in a tuple of Exceptions).